### ISSUE TITLE
Don't register berkelydb as a store if it is not available on the system

### ISSUE THREAD

**Issue by aucampia** [RDFLib/rdflib#1816]
Currently we are always registering BerkelyDB regardless of whether it is available or not:

https://github.com/RDFLib/rdflib/blob/cdaee27415bd669082d90c9b6bbff3191a147761/rdflib/plugin.py#L202-L207

This causes problems if we want to operate on all stores, for example to test some functionality, as a special exception is needed for BerkeleyDB:

https://github.com/RDFLib/rdflib/blob/13ec1df32bdb6a09de491ba2a931008c9dd9adea/test/test_graph/test_graph.py#L68-L84

I think it would be best to only register the plugin if the store is available.

**Comment by Jtmcflea** [comment]
Gonna try working on this, would adding some kind of availability check in the `register` method work?

**Comment by Jtmcflea** [comment]
Also, I'm assuming the method should check if all plugins are available. Should I do that or just have a specific check for berkelydb?

**Comment by aucampia** [comment]
> Gonna try working on this, would adding some kind of availability check in the `register` method work?

> Also, I'm assuming the method should check if all plugins are available. Should I do that or just have a specific check for berkelydb?

Hi @Jtmcflea I think first and most critical is to just not register plugins we know won't work, sadly there is no generic check we can do for all plugins, we know berkeleydb won't work unless we can import `berkeleydb` - but berkelydb is really an exception here, not the rule. Most (all?) other stores will work fine with stock python. For other plugins like https://github.com/RDFLib/rdflib-sqlalchemy there may be similar considerations, but that is not something to worry about in this repo.

So I would suggest just do it for berkeleydb.

**Comment by Jtmcflea** [comment]
Ok, I've moved the berkeleydb register to the top of the list and added `if rdflib.plugins.stores.berkeleydb.has_bsddb:` before it to check if the system has berkeleydb. I've also removed the exception in test_graph.py, which passed, and I'm currently running the validate action in github.

Any more tests or formatting things I need to do? Sorry to ask, but I'm new at open source contribution in general.
